packages/world/src/modules/init/implementations/StoreRegistrationSystem.sol

```diff
@@ -44,8 +45,8 @@ contract StoreRegistrationSystem is System, IWorldErrors, LimitedCallContext {
     FieldLayout fieldLayout,
     Schema keySchema,
     Schema valueSchema,
-    string[] calldata keyNames,
-    string[] calldata fieldNames
```

Comment:
what if we instead change it at the system lib level so we can keep the existing systems as they are?

Reply:
as in turn `calldata` into `memory` for just system lib methods?

Reply:
fwiw I saw no change in gas for this change, what's the concern here?

Reply:
yeah, gas improvements from using calldata will only happen if you don't turn those args into memory later, which most of these systems might be doing anyways. But imo it's better to have them in case a system would benefit from it in the future, also the api remains the same.

Reply:
done